src/pages/tasks/common/hooks/useInvoiceTask.ts

```diff
@@ -55,7 +64,30 @@ export function useInvoiceTask() {
     return hoursSum;
   };
 
-  return (tasks: Task[]) => {
+  const fetchProjectDetails = async (projectId: string) => {
+    let project: Project | undefined;
+
+    await queryClient
+      .fetchQuery(generatePath('/api/v1/projects/:id', { id: projectId }), () =>
+        request(
+          'GET',
+          endpoint('/api/v1/projects/:id', {
+            id: projectId,
+          })
+        )
+      )
+      .then((response: GenericSingleResourceResponse<Project>) => {
+        project = response.data.data;
+      })
+      .catch((error: AxiosError) => {
+        toast.error();
+        console.error(error);
+      });
+
+    return project;
+  };
```

Comment:
Shouldn't we extract this to hook as other queries?

Reply:
Hmm, tbh I've looked at other cases where we use `queryClient.fetchQuery` and nowhere do we separate that in the hook. We just created a function and in that function we called `fetchQuery` (or we call it in the useEffect). We have that case on the `invoicing project`, there we only have the function `handleInvoiceProject` where we call fetchQuery. IMO, it's good to keep it like this.

But, of course, if you think it's much better to extract it, I'll do it. Let me know your thoughts.

Reply:
@beganovich Aham, okay. I extracted it into hook. I hope that you agree with this solution. Let me know your thoughts.